packages/wallet/backend/src/socket/service.ts

```diff
@@ -0,0 +1,63 @@
+import { env, type Env } from '@/config/env'
+import { Logger } from 'winston'
+import socketIo from 'socket.io'
+import http from 'http'
+import { unsealData } from 'iron-session'
+import { AccountService } from '@/account/service'
+
+interface ISocketService {
+  init(args: http.Server): void
+  emitAccountsUpdateById(id: string): void
+}
+
+interface SocketServiceDependencies {
+  logger: Logger
+  env: Env
+  accountService: AccountService
+}
+
+export class SocketService implements ISocketService {
+  private io: socketIo.Server | null = null
+
+  constructor(private deps: SocketServiceDependencies) {}
+
+  init(server: http.Server) {
+    this.io = new socketIo.Server(server, {
+      cors: {
+        origin: [
+          'http://localhost:4003',
+          `https://${this.deps.env.RAFIKI_MONEY_FRONTEND_HOST}`
+        ],
+        credentials: true
+      }
+    })
+    this.deps.logger.info(`Socket Server is started...`)
+
+    this.io.on('connection', async (socket) => {
+      const token = socket.handshake.query.token?.toString().split('=')[1]
```

Comment:
```suggestion
      const token = socket.handshake.auth.token?.toString().split('=')[1]
```

Reply:
by using the auth, it will be sent as a header @dragosp1011 on the initial HTTP upgrade request

Reply:
agree with this approach, looks like it is the recommended way